packages/firestore-types/index.d.ts

```diff
@@ -208,7 +225,9 @@ export class DocumentReference<T = DocumentData> {
 
   isEqual(other: DocumentReference<T>): boolean;
 
-  set(data: T, options?: SetOptions): Promise<void>;
+  set(data: Partial<T>, options: SetOptions): Promise<void>;
+  set(data: T): Promise<void>;
+  set(data: T | Partial<T>, options?: SetOptions): Promise<void>;
```

Comment:
This catch-all override should not be needed in the type file.

Reply:
removed.